### PR TITLE
Toggle Visibility of Template Control Bar

### PR DESCRIPTION
**Pull request by alexpaxton**
- [x] CHANGELOG.md updated
  - [x] Rebased/mergable
  - [x] Tests pass

Connect #1419 

### The problem
- Template Control Bar is always visible

### The Solution
- Add button in dash header to toggle the visibility of Template Control Bar
- Added an empty state to Template Control Bar
- Persist open/closed state of Template Control Bar
- Hide Template Control Bar when in presentation mode

### Preview
template-control-bar

### CODE REVIEW

**Comment by jaredscheib** [comment]
So this means that if a user has more than 2 rows of Template Variables, they will not show?

**Comment by alexpaxton** [comment]
@jaredscheib removed the animations & height limitations, @cryptoquick said he would help persist this state to localstorage (which seems like the best option)

**Comment by jaredscheib** [comment]
Nice. Tested it -- looks good. I can help with the redux localStorage stuff.

Question: do we want the TVM to stick to the header as you scroll down a dashboard?
screen shot 2017-05-22 at 3 04 55 pm
screen shot 2017-05-22 at 3 04 44 pm

**Comment by alexpaxton** [comment]
Note: made a UX decision that showing/hiding the tempvars control bar should be persisted

TODO: Write test for the reducer

**Comment by jaredscheib** [comment]
let's also hide the TV Controls when in presentation mode. and note my question
above @alexpaxton

File: ui/src/dashboards/components/DashboardHeader.js [hunk 1]
```diff
@@ -59,6 +62,16 @@ const DashboardHeader = ({
                   Rename
                 </button>
               : null}
+            {dashboard
```
Comment (lukevmorris) [inline ui/src/dashboards/components/DashboardHeader.js]: At some point I think we need to refactor this and other Dashboard components with the assumption that there will always be a Dashboard present. I don't like having to perform this check all the time

Comment (jaredscheib) [inline ui/src/dashboards/components/DashboardHeader.js]: good call @lukevmorris. could you check https://github.com/influxdata/chronograf/issues/1372 and add/amend where necessary? that's the running list of these kinds of refactoring.

Comment (lukevmorris) [inline ui/src/dashboards/components/DashboardHeader.js]: Added a bullet point at the end

**Pull request merged**